README.md

```diff
@@ -2,63 +2,101 @@
 
-You may test the package as follows:
+```julia
+import Pkg; Pkg.test("GigaSOM");
+```
 
+For debugging, it is sometimes very useful to enable the `@debug` messages from the source, as such:
 ```julia
-julia> import Pkg; Pkg.test("GigaSOM");
+using Logging
+global_logger(ConsoleLogger(stderr, Logging.Debug))
 ```
 
-Alternatively, you can use the package manager by hitting `]`:
+## How to use GigaSOM
+
+A comprehensive documentation is [available online](https://lcsb-biocore.github.io/GigaSOM.jl/latest/); several [introductory tutorials](https://lcsb-biocore.github.io/GigaSOM.jl/latest/tutorials/basicUsage/) of increasing complexity are also included.
+
+A very basic dataset (Levine13 from [FR-FCM-ZZPH](https://flowrepository.org/id/FR-FCM-ZZPH)) can be loaded, clustered and visualized as such:
 
 ```julia
-(v1.1) pkg> test GigaSOM
+using GigaSOM
+
+params, fcsmatrix = loadFCS("Levine_13dim.fcs")  # load the FCS file
+
+exprs = fcsmatrix[:,1:13]  # extract only the data columns with expression values
+
+som = initGigaSOM(exprs, 20, 20)    # random initialization of the SOM codebook
+som = trainGigaSOM(som, exprs)      # SOM training
+clusters = mapToGigaSOM(som, exprs) # extraction of per-cell cluster IDs
+e = embedGigaSOM(som, exprs)        # EmbedSOM projection to 2D
 ```
 
-For debugging, it may be useful to enable the `@debug` messages from source, in the simplest case by:
+The example loads the data, runs the SOM training (as in FlowSOM) and computes a 2D projection of the dataset (using EmbedSOM); the total computation time (excluding the possible precompiltion of the libraries) should be around 15 seconds.
+
+The results can be visualized e.g. with [GigaScatter](https://github.com/LCSB-BioCore/GigaScatter.jl#usage-with-gigasomjl) which we developed for this purpose, or by exporting the data and plotting them with any other programming language. For example, to save an embedding with highlighted expression of CD4, you can install and use GigaScatter as such:
+
 ```julia
-using Logging
-global_logger(ConsoleLogger(stderr, Logging.Debug))
+import Pkg; Pkg.add("GigaScatter")
+using GigaScatter
+
+savePNG("Levine13-CD4.png",
+  solidBackground(rasterize((500,500),        # bitmap size
+    Matrix{Float64}(e'),                      # the embedding coordinates
+    expressionColors(
+      scaleNorm(Array{Float64}(exprs[:,5])),  # 5th column contains CD4 expressions
+      expressionPalette(100, alpha=0.5)))))   # colors for plotting (based on RdYlBu)
 ```
 
-:warning: It is not advised to run indivual test files separately without
-expliciting activating the environment and loading the dependencies.  If this
-is required for debugging purposes, please activate the environment first and
-run the commands of the `test/runtests.jl` file sequentially.
+The output may look like this (blue is negative expresison, red is positive):
+
+![Levine13 embedding with CD4 highlighted](media/Levine13-CD4.png "Levine13/C4")
+
+
+## Feedback, issues, questions
```

Comment:
I would refer here to the contributing guide, this could eventually even be integrated there.

Reply:
OK, will change together with that.

Reply:
(Eh, I can't push to the branch with CoC and contribution guide, can you just link it there after this gets merged?)